python/tvm/contrib/debugger/debug_result.py

```diff
@@ -0,0 +1,207 @@
+        to construct a "PackedFunc" .
+
+    dump_path : str
+        Output data path is read/provided from frontend
+    """
+
+    def __init__(self, graph_json, dump_path):
+        self._dump_path = dump_path
+        self._output_tensor_list = []
+        self._time_list = []
+        self._parse_graph(graph_json)
+        # dump the json information
+        self.dump_graph_json(graph_json)
+
+    def _parse_graph(self, graph_json):
+        """Parse and extract the NNVM graph and update the nodes, shapes and dltype.
+
+        Parameters
+        ----------
+        graph_json : str or graph class
+           The graph to be deployed in json format output by nnvm graph.
+        """
+        json_obj = json.loads(graph_json)
+        self._nodes_list = json_obj['nodes']
+        self._shapes_list = json_obj['attrs']['shape']
+        self._dtype_list = json_obj['attrs']['dltype']
+        self._update_graph_json()
+
+    def _update_graph_json(self):
+        """update the nodes_list with name, shape and data type,
+        for temporarily storing the output.
+        """
+
+        nodes_len = len(self._nodes_list)
+        for i in range(nodes_len):
+            node = self._nodes_list[i]
+            input_list = []
+            for input_node in node['inputs']:
+                input_list.append(self._nodes_list[input_node[0]]['name'])
+            node['inputs'] = input_list
+            dtype = str("type: " + self._dtype_list[1][i])
+            if 'attrs' not in node:
+                node['attrs'] = {}
+                node['op'] = "param"
+            else:
+                node['op'] = node['attrs']['func_name']
+            node['attrs'].update({"T": dtype})
+            node['shape'] = self._shapes_list[1][i]
+
+    def _cleanup_tensors(self):
+        """Remove the tensor dump file (graph wont be removed)
+        """
+        for filename in os.listdir(self._dump_path):
+            if os.path.isfile(filename) and not filename.endswith(".json"):
+                os.remove(filename)
+
+    def get_graph_nodes(self):
+        """Return the nodes list
+        """
+        return self._nodes_list
+
+    def get_graph_node_shapes(self):
+        """Return the nodes shapes list
+        """
+        return self._shapes_list
+
+    def get_graph_node_output_num(self, node):
+        """Return the number of outputs of a node
+        """
+        return 1 if node['op'] == 'param' else int(node['attrs']['num_outputs'])
+
+    def get_graph_node_dtypes(self):
+        """Return the nodes dtype list
+        """
+        return self._dtype_list
+
+    def set_output_tensor(self, out_tensor):
+        """Set the output tensor of a node to the list, the list is appended in the same order as
+        node list
+
+        Parameters
+        ----------
+        out_tensor : ndarray
+            The output tensor in ndarray format
+        """
+        self._output_tensor_list.append(out_tensor)
+
+    def set_time(self, time):
```

Comment:
this seems to be too simple function to add a level of indirection, directly inline at call site

Reply:
+1